src/main/scala/faces/utils/Helpers.scala

```diff
@@ -287,15 +290,398 @@ case class Helpers(cfg: FacesSettings)(implicit rnd: Random) {
+      start_x = r.nextInt(cfg.imageDimensions.imageWidth - 2) + 1
+      start_y = r.nextInt(cfg.imageDimensions.imageHeight - 2) + 1
+      height = r.nextInt(cfg.imageDimensions.imageHeight - 1 - start_y)
+      current_pixels = 0
+      var current_x = start_x
+      while (current_pixels < target_pixels && current_x < mask.width - 1) {
+        current_pixels += countFacePixelsInCol(mask, current_x, height)
+        current_x += 1
+      }
+      width = current_x - start_x
+      if (current_pixels >= target_pixels)
+        ok = true
+    }
+    (start_x, start_y, width, height)
+  }
+
+  def countFacePixelsInCol(mask: PixelImage[RGBA], col: Int, height: Int): Int = {
+    var counter = 0
+    for (i <- 0 to height) {
+      if (mask.apply(col, i) == RGBA.White) {
+        counter += 1
+      }
+    }
+    counter
+  }
+
+  def countFacePixels(img: PixelImage[RGBA]): Long = {
+    var counter = 0
+    for (i <- 0 to img.width - 1) {
+      for (j <- 0 to img.height - 1) {
+        if (img(i, j) == RGBA.White)
+          counter += 1
+      }
+    }
+    counter
+  }
+
+
+  def writeOcclusionToImages(image: PixelImage[RGBA], mask: PixelImage[RGBA], occlusion: PixelImage[RGBA], start: (Int, Int)): (PixelImage[RGBA], PixelImage[RGBA]) = {
+    val return_Image = PixelImage(image.width, image.height, (x, y) => {
+      if (x >= start._1 && x < start._1 + occlusion.width && y >= start._2 && y < start._2 + occlusion.height) {
+        val occlusion_rgba = occlusion(x - start._1, y - start._2)
+        if (occlusion_rgba.a > 0.5)
+          occlusion(x - start._1, y - start._2)
+        else
+          image(x, y)
+      }
+      else
+        image(x, y)
+    })
+    val return_Mask = PixelImage(mask.width, mask.height, (x, y) => {
+      if (x >= start._1 && x < start._1 + occlusion.width && y >= start._2 && y < start._2 + occlusion.height) {
+        val occlusion_rgba = occlusion(x - start._1, y - start._2)
+        if (occlusion_rgba.a > 0.5)
+          RGBA.apply(0, 0, 0, 0) // Change the color of the occlusion on the GROTRU mask
+        else
+          mask(x, y)
+      }
+      else
+        mask(x, y)
+    })
+    (return_Image, return_Mask)
+  }
+
+  def flipImage(image: PixelImage[RGBA], flip: (Boolean, Boolean, Boolean)): PixelImage[RGBA] = {
+    if (flip._3)
+      PixelImage(image.height, image.width, (x, y) => {
+        var x_new = x
+        var y_new = y
+        if (flip._1)
+          x_new = image.height - 1 - x
+        if (flip._2)
+          y_new = image.width - 1 - y
+        image(y_new, x_new)
+      })
+    else
+      PixelImage(image.width, image.height, (x, y) => {
+        var x_new = x
+        var y_new = y
+        if (flip._1)
+          x_new = image.width - 1 - x
+        if (flip._2)
+          y_new = image.height - 1 - y
+        image(x_new, y_new)
+      })
+  }
+
+  def getRandImage(onlyHands: Int): (Boolean, Boolean, Boolean, String, Int) = {
+    val r = scala.util.Random
+    val A = Array(
```

Comment:
please don't use hardcoded names here - read the directory instead. I think it would make sense to have hands and micros in different directories and then read the whole folder (see bip or background implementation)